### PR TITLE
Add possibility to disable service account creation for backup servic…

### PR DESCRIPTION
**Pull request by bartlomiejsawicki**
…e (it is required when use externally created SA)

Signed-off-by: Bartłomiej Sawicki <mechu@mechu.net>

### What this PR does / why we need it
Add possibility to disable service account creation for backup service (it is required when use externally created SA)

### Checklist

- [x] [DCO](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] CHANGELOG.md was updated

### CODE REVIEW

File: charts/jenkins/templates/jenkins-backup-rbac.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {{- if .Values.backup.enabled }}
+  {{- if .Values.backup.createServiceAccount }}
```
Comment (timja) [inline charts/jenkins/templates/jenkins-backup-rbac.yaml:2]:
```suggestion
{{- if .Values.backup.createServiceAccount }}
```

File: charts/jenkins/templates/jenkins-backup-rbac.yaml [hunk 2]
```diff
@@ -16,6 +17,7 @@ metadata:
   annotations:
     {{- toYaml .Values.backup.annotations | nindent 4 }}
   {{- end }}
+  {{- end }}
```
Comment (timja) [inline charts/jenkins/templates/jenkins-backup-rbac.yaml:20]:
```suggestion
{{- end }}
```

File: charts/jenkins/values.yaml [hunk 3]
```diff
@@ -799,6 +799,8 @@ backup:
   # Ref: https://crontab.guru/
   schedule: "0 2 * * *"
   labels: {}
+  # Specifies whether a ServiceAccount should be created
```
Comment (timja) [inline charts/jenkins/values.yaml:802]:
```suggestion
```

**Review by timja — COMMENTED** [review]
looks good, I think some minor whitespace cleanup should be done

**Review by timja — APPROVED** [review]

**Comment by torstenwalter** [comment]
Do you also need to specify the name of the service account?

If yes the it would be worth a try to align namings with the other service account values.

```
serviceAccount.create
serviceAccount.name

serviceAccountAgent.create
serviceAccountAgent.name
```

So maybe
```
serviceAccountBackup.create
serviceAccountBackup.create
```

or
```
backup.serviceAccount.create
backup.serviceAccount.name
```

The name can also be used in the backup job to specify if the existing service account has a different name.

See also https://v2.helm.sh/docs/chart_best_practices/#yaml-configuration

**Review by torstenwalter — APPROVED** [review]
Looks good! 💯
Thank you!

**Pull request merged**